combat-trainer.lic

```diff
@@ -138,7 +138,19 @@ class SetupProcess
       echo('skipping summoned weapons because no moonblade available') if $debug_mode_ct
       weapon_training = weapon_training.reject { |skill, _| game_state.summoned_info(skill) }
     end
-    new_weapon_skill = weapon_training.min_by { |skill, _| [DRSkill.getxp(skill), DRSkill.getrank(skill)] }.first
+
+    new_weapon_skill=""
+    echo "Priority weapons: #{get_settings.priority_weapons}" if $debug_mode_ct
+    get_settings.priority_weapons.each do | weapon |
+      if DRSkill.getxp(weapon)==0 && weapon_training[weapon]
```

Comment:
You could use method chaining instead of putting these conditionals in the `.each`

Reply:
I'm not sure how to do that.  Can you demonstrate?

Reply:
I think he means something like this? 

```new_weapon_skill = get_settings.priority_weapons.select{|weapon| weapon_training[weapon] && DRSkill.getxp (weapon)}```

Reply:
```array.flatten.compact.uniq``` type of thing

Reply:
Something like:
```
    get_settings.priority_weapons
    .select { |weapon| DRSkill.getxp(weapon)==0 }
    .select { |weapon| weapon_training[weapon] }
    .each { | weapon | new_weapon_skill=weapon }
```

Although you'll want to account for some of the other comments (like [this](https://github.com/rpherbig/dr-scripts/pull/2240/files/66a604e2cde436ffa059594c2c4a250115c76b13#r157391826) and [this](https://github.com/rpherbig/dr-scripts/pull/2240/files/66a604e2cde436ffa059594c2c4a250115c76b13#r157391904)).